txscript/sighash.go

```diff
@@ -626,3 +626,20 @@ func CalcTapscriptSignaturehash(sigHashes *TxSigHashes, hType SigHashType,
 		sigHashes, hType, tx, idx, prevOutFetcher, opts...,
 	)
 }
+
+// SigHashTypeForScript returns the sighash flag for the given UTXO's pkScript.
+func SigHashTypeForScript(script []byte) (SigHashType, error) {
```

Comment:
Hmm, I'm not sure how generally useful this is: it 100% assumes that a given script will use sighash all for taproot. Also there isn't necessarily a direct link between a script and the type of sighash it'll use. This would typically be dependent on the application context.

Reply:
`This would typically be dependent on the application context.` <= my fault. I thought there as a direct (or at least in "almost all cases") link between sighash <=> scriptype